app/browser/reducers/passwordManagerReducer.js

```diff
@@ -86,178 +87,145 @@ const init = () => {
-      }
-      return
-    }
+  const webContents = getWebContents(tabId)
 
-    masterKey = masterKey || getMasterKey()
-    if (!masterKey) {
-      console.log('Could not access master password; aborting')
-      return
-    }
+  passwordCallbacks[message] = (buttonIndex) => {
+    delete passwordCallbacks[message]
+    appActions.hideNotification(message)
 
-    results.forEach((result) => {
-      usernames[result.get('username')] = CryptoUtil.decryptVerify(result.get('encryptedPassword'),
-                                                                   result.get('authTag'),
-                                                                   masterKey,
-                                                                   result.get('iv')) || ''
-    })
-    let win = BrowserWindow.getFocusedWindow()
-    if (!win) {
+    if (buttonIndex === 0) {
+      webContents.updatePassword()
       return
     }
-    if (Object.keys(usernames).length > 0) {
-      win.webContents.send(messages.SHOW_USERNAME_LIST,
-                           usernames, origin, action,
-                           boundingRect)
-    } else {
-      win.webContents.send(messages.HIDE_CONTEXT_MENU)
-    }
-  })
-
-  ipcMain.on(messages.SAVE_PASSWORD, (e, username, password, origin, action) => {
-    if (!password || !origin || !action) {
-      return
-    }
-    const originSettings = siteSettings.getSiteSettingsForURL(AppStore.getState().get('siteSettings'), origin)
-    if (originSettings && originSettings.get('savePasswords') === false) {
-      return
-    }
-
-    masterKey = masterKey || getMasterKey()
-    if (!masterKey) {
-      console.log('Could not access master password; aborting')
-      return
-    }
-
-    const passwords = AppStore.getState().get('passwords')
+    // never save
+    webContents.noUpdatePassword()
+  }
+}
 
-    // If the same password already exists, don't offer to save it
-    let result = passwords.findLast((pw) => {
-      return pw.get('origin') === origin && pw.get('action') === action && (username ? pw.get('username') === username : !pw.get('username'))
+const migrate = (state) => {
+  masterKey = masterKey || getMasterKey()
+  if (!masterKey) {
+    console.log('Could not access master password; aborting')
+    return
+  }
+  const passwords = state.get('passwords')
+  if (passwords.size) {
+    passwords.forEach((password) => {
+      let decrypted = CryptoUtil.decryptVerify(password.get('encryptedPassword'),
+                                               password.get('authTag'),
+                                               masterKey,
+                                               password.get('iv'))
+      let form = {}
+      form['origin'] = password.get('origin')
+      form['signon_realm'] = password.get('origin') + '/'
+      form['action'] = password.get('action')
+      form['username'] = password.get('username')
+      form['password'] = decrypted
```

Comment:
It will be saved in our user dir as `Login Data` sqlite format encrypted by OS keychain.
So for example if you are on Mac, Brave will prompt to ask keychain permission.